test/integration/framework/shoot_app.go

```diff
@@ -0,0 +1,585 @@
+	kubeconfig, err := getObjectFromSecret(ctx, o.K8sSeedClient, o.Shoot.SeedNamespace, v1beta1.GardenerName, kubeconfig)
+	if err != nil {
+		return "", err
+	}
+
+	if len(downloadPath) != 0 {
+		file, err := os.Create(downloadPath)
+		if err != nil {
+			return "", err
+		}
+		defer func() {
+			err := file.Close()
+			if err != nil {
+				o.ShootGardenerTest.Logger.Fatal("Could not close file")
+			}
+		}()
+
+		_, err = file.WriteString(kubeconfig)
+		if err != nil {
+			return "", err
+		}
+	}
+	return kubeconfig, nil
+}
+
+// DashboardAvailable checks if the dashboard is available
+func (o *GardenerTestOperation) DashboardAvailable(ctx context.Context) error {
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	httpClient := http.Client{
+		Transport: transport,
+		Timeout:   time.Duration(5 * time.Second),
+	}
+
+	url := fmt.Sprintf("https://api.%s/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy", *o.Shoot.Info.Spec.DNS.Domain)
+	httpRequest, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+
+	dashboardPassword, err := o.getAdminPassword(ctx)
+	if err != nil {
+		return err
+	}
+
+	httpRequest.SetBasicAuth("admin", *dashboardPassword)
+	r, err := httpClient.Do(httpRequest)
+	if err != nil {
+		return err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("dashboard unavailable")
+	}
+
+	return nil
+}
+
+// HTTPGet performs an HTTP GET request with context
+func (o *GardenerTestOperation) HTTPGet(ctx context.Context, url string) (*http.Response, error) {
+	httpClient := http.Client{}
+	httpRequest, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	httpRequest = httpRequest.WithContext(ctx)
+	response, err := httpClient.Do(httpRequest)
+	return response, err
+}
+
+// DeployTiller creates a tiller deployment on the shoot cluster
+func (o *GardenerTestOperation) DeployTiller(ctx context.Context) error {
+	tillerDeployment, err := o.generateDeploymentObject(o.TillerConfigurationOptions)
+	if err != nil {
+		return err
+	}
+	var deploymentToReturn *appsv1.Deployment
+	err = o.K8sShootClient.Client().Get(ctx, types.NamespacedName{Namespace: tillerDeployment.Namespace, Name: tillerDeployment.Name}, deploymentToReturn)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = o.K8sShootClient.Client().Create(ctx, tillerDeployment)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+// WaitForDeploymentToComplete waits for the shoot to be created
+func (o *GardenerTestOperation) WaitForDeploymentToComplete(timeoutCtx context.Context, deployment *appsv1.Deployment, pollInterval time.Duration) error {
+	if o.TillerConfigurationOptions == nil {
+		return ErrEmptyTillerOptionsConfiguration
+	}
+
+	return wait.PollImmediateUntil(pollInterval, func() (bool, error) {
+		ok := make(chan bool)
+		errChannel := make(chan error)
+
+		go func() {
```

Comment:
You don't gain anything by using a goroutine here since you block anyways at the end - The remaining `ctx` handling is anyways already done by `PollImediateUntil` -> Remove goroutine inside the `PollImmediateUntil` and also set a constant value for your `pollInterval`, the caller of this thing shouldn't have to decide on that.